birdhouse/config/jupyterhub/config/magpie/docker-compose-extra.yml

```diff
@@ -1,5 +1,7 @@
 version: "3.4"
 services:
   magpie:
+    environment:
+      MAGPIE_USER_NAME_EXTRA_REGEX: ^\S+$$
```

Comment:
I am completely confused here.  So `\S` will "matches any non-whitespace character; this is equivalent to the class [^ \t\n\r\f\v]." (definition from https://docs.python.org/3/howto/regex.html#simple-patterns).

So we want the usernames to only contain non-writespace char ???

Other than non-whitespace, we probably would want to avoid other kind of illegal char like `/`, `*`, `#` ...  Unless this is already handled by Magpie built-in username validation?

Or I might have completely mis-understood how the new `MAGPIE_USER_NAME_EXTRA_REGEX` works.

Reply:
Opps sorry, my brain is dead past midnight, yes we want non-whitespace char.  

The regex is right but I'd like something more conservative, like an explicit allow list: `"^[a-zA-Z0-9_\-]+$$"`.  This explicit list will protect us from any other illegal char that we ever forgot to think about.  I'd rather be safe than sorry.

Reply:
This regex is used in combination with the one already enforced by Magpie by default `^[A-Za-z0-9]+(?:[\s_\-\.][A-Za-z0-9]+)*$` (see https://github.com/bird-house/birdhouse-deploy/pull/396#pullrequestreview-1698302553).

The username must match both that default regex as well as `^\S$`. The default regex protects us from unexpected characters, the second regex protects us from whitespace.

I'd rather not duplicate most of the default regex here since the more complex we make it, the more prone to errors we are if we change it in the future.

Reply:
If we are going to override the regex either way, I agree with @tlvu that it is better to have it explicitly, even if they overlap partially. It is easier to analyze one location than having to go back into Magpie code to validate the full set. Even if we do make a mistake, the regex on Magpie side will still double-check it.